packages/stream_chat/lib/src/client/channel.dart

```diff
@@ -258,8 +258,34 @@ class Channel {
     return state!.channelStateStream.map((cs) => cs.channel?.cooldown ?? 0);
   }
 
+  /// Remaining cooldown duration in seconds for the channel.
+  ///
+  /// Returns 0 if there is no cooldown active.
+  int get remainingCooldown {
+    _checkInitialized();
+
+    final cooldownDuration = cooldown;
+    if (cooldownDuration <= 0) return 0;
+
+    final userLastMessageAt = currentUserLastMessageAt;
+    if (userLastMessageAt == null) return 0;
+
+    if (ownCapabilities.contains(PermissionType.skipSlowMode)) return 0;
+
+    final currentTime = DateTime.timestamp();
+    final elapsedTime = currentTime.difference(userLastMessageAt).inSeconds;
+
+    return max(0, cooldownDuration - elapsedTime);
+  }
+
   /// Stores time at which cooldown was started
-  DateTime? cooldownStartedAt;
+  @Deprecated(
+    "Use a combination of 'remainingCooldown' and 'currentUserLastMessageAt'",
+  )
```

Comment:
Is there really a need to deprecate this getter? It still works as expected right? And there isn't a lot to maintain here.

Reply:
This isn't being used anymore and has no specific use, it was a SDK specific hack created earlier to store the cooldown start time so that we can resume it when we go back to the channel. This however was not enough to also resume it if we are restarting the app.

Reply:
I don't know if anybody uses this, but I agree it's probably not a very useful method.